internal/trie/node/decode_test.go

```diff
@@ -333,7 +336,8 @@ func Test_decodeLeaf(t *testing.T) {
 			variant:          leafVariant.bits,
 			partialKeyLength: 1,
 			errWrapped:       ErrDecodeValue,
-			errMessage:       "cannot decode value: could not decode invalid integer",
+			errMessage: "cannot decode value: unmarshal failed: decodeBytes failed: decodeLength failed: " +
```

Comment:
This is a tiny out of scope, but essentially I was wrapping `ErrDecodeValue` using the scale error string as context since I couldn't use any wrapped error from scale before. Now that this is fixed in this PR, perhaps you could remove the `ErrDecodeValue`, replace its usage with a string context `decoding value: ` and update the `errWrapped` in tests to match the corresponding `scale.ErrXXX`?